enacts/wat_bal/agronomy.py

```diff
@@ -214,3 +214,105 @@ def soil_plant_water_balance(
             - (planted_since - np.timedelta64(1, "D"))
         )
     return sm, drainage, et_crop, planting_date
+
+
+def weekly_api_runoff(
+    daily_rain,
+    no_runoff=12.5,
+    api_thresh=xr.DataArray([6.3, 19, 31.7, 44.4, 57.1, 69.8], dims=["api_cat"]),
+    api_poly=xr.DataArray(
+        [
+            [0.858, 0.0895, 0.0028],
+            [-1.14, 0.042, 0.0026],
+            [-2.34, 0.12, 0.0026],
+            [-2.36, 0.19, 0.0026],
+            [-2.78, 0.25, 0.0026],
+            [-3.17, 0.32, 0.0024],
+            [-4.21, 0.438, 0.0018],
+        ],
+        dims=["api_cat", "powers"],
+    ),
+    time_dim="T",
+):
+    """Computes Runoff using Antecedent Precipitation Index.
+    `runoff` is a polynomial of `daily_rain` of order 2.
+    Polynomial is chosen based on API categories.
+    Additionaly, `runoff` is 0 if it rains less or equal than `no_runoff` ,
+    and negative `runoff` is 0.
+    Parameters
+    ----------
+    daily_rain : DataArray
+        daily precipitation
+    no_runoff : DataArray, optional
+        `runoff` is 0 if `daily_rain` is leser or equal to `no_runoff`
+        (default `no_runoff` =12.5)
+    api_thresh : DataArray, optional
+        increasing daily API values along a dimension called api_cat
+        indicating the upper limit (inlcusive) to belong to an API category
+    api_poly : DataArray, optional
+        polynomial coefficients that must depend on a dimension called powers
+        of size 3 (the 3 powers of a polynomial of order 2),
+        and on a dimension api_cat of size one more than `api_thresh` 's size.
+        The polynomial used to compute the `runoff` is picked according to the categories
+        defined by the thresholds.
+    time_dim : str, optional
+        daily time dimension of `daily_rain` (default `time_dim` ="T").
+        
+    Returns
+    -------
+    runoff : DataArray
+        daily Runoff.
+    See Also
+    --------
+    api_sum
+    Notes
+    -----
+    For instance with the default parameters, if rain is greater or equal to 12.5
+    and API is 18, then runoff is -1.14 + 0.042*x 0.0026*(x**2)
```

Comment:
Right... Should not those default values be in the documentation actually? In which case it will be visible to the doc reader.

Runoff < 0 set to 0 is documented in the first paragraph of the doc already.

Reply:
oh I didn't think of that: I thought I should write them down. Either way.

Reply:
It does happen automatically even though it's not pretty.

Reply:
Resolve this unless you want a reminder to prettify the documentation.